aiohue/v2/__init__.py

```diff
@@ -246,3 +247,32 @@ async def fetch_full_state(self) -> None:
             self._sensors.initialize(full_state),
             self._groups.initialize(full_state),
         )
+
+    async def get_diagnostics(self) -> Dict[str, Any]:
+        """Return a dict with diagnostic information for debugging and support purposes."""
+        result = {}
+        subst_ids = {}
+
+        def _replace_id(item: dict):
+            """Replace ID with a random one for privacy reasons."""
+            id_key = "id" if "id" in item else "rid"
+            org_item_id = item[id_key]
+            new_id = subst_ids.get(org_item_id, f"redacted-{uuid4()}")
+            subst_ids[org_item_id] = new_id
+            item[id_key] = new_id
+
+        # add full state to result
+        full_state = await self.request("get", "clip/v2/resource")
+        for item in full_state:
+            _replace_id(item)
+
+        result["full_state"] = full_state
+
+        # add last event messages to result
+        last_events = list(self._events.last_events)
+        for item in last_events:
```

Comment:
No need to copy the list. With async there is just 1 task running at a time.
```suggestion
        for item in self._events.last_events:
```